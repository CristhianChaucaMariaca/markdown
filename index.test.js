const createParser = require('./index.js');


const parser = createParser();


const markdown = `![Platzi Logo](https://static.platzi.com/static/images/logos/platzi@2x.png)

*[HTML]: Hyper Text Markup Language
*[W3C]:  World Wide Web Consortium

# Heading 1
Lorem **ipsum** dolor _sit_ ++amet++.

The HTML ==specification== is maintained by the W3C.

- \`item 1\`
- item 2
- item 3

Term 1
  ~ Definition 1

Term 2
  ~ Definition 2a
  ~ Definition 2b

\`\`\`js
function greeting() {
	return 'hello world!';
}
\`\`\`

@[youtube](ajLJOhf-WdA)`;


test('Platzi Flavored Markdown parser', () => {
  const html = parser(markdown);
  expect(html).toMatchSnapshot();
});
