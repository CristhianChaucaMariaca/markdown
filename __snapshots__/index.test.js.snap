exports[`test Platzi Flavored Markdown parser 1`] = `
"<figure><img src=\"https://static.platzi.com/static/images/logos/platzi@2x.png\" alt=\"Platzi Logo\" /></figure>
<h1>Heading 1</h1>
<p>Lorem <strong>ipsum</strong> dolor <em>sit</em> <ins>amet</ins>.</p>
<p>The <abbr title=\"Hyper Text Markup Language\">HTML</abbr> <mark>specification</mark> is maintained by the <abbr title=\"World Wide Web Consortium\">W3C</abbr>.</p>
<ul>
<li><code>item 1</code></li>
<li>item 2</li>
<li>item 3</li>
</ul>
<dl>
<dt>Term 1</dt>
<dd>Definition 1</dd>
<dt>Term 2</dt>
<dd>Definition 2a</dd>
<dd>Definition 2b</dd>
</dl>
<pre><code class=\"language-js\">function greeting() {
	return \'hello world!\';
}
</code></pre>
<p><div class=\"embed-responsive embed-responsive-16by9\"><iframe class=\"embed-responsive-item\" id=\"youtubeplayer\" type=\"text/html\" width=\"640\" height=\"390\" src=\"//www.youtube.com/embed/ajLJOhf-WdA\" frameborder=\"0\" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div></p>
"
`;
